include/wil/win32_helpers.h

```diff
@@ -20,7 +20,7 @@
 
 // detect std::bit_cast
 #ifdef __has_include
-#  if __has_include(<bit>)
+#  if (__cplusplus > 202000) && __has_include(<bit>)

```

Comment:
The original PR was like that (#168) but it was changed to the current version by https://github.com/microsoft/wil/pull/174/commits/e2afbc74e9ec3831242ada4fcf1141ebc8bd0d6e. Not sure what the build issue that the commit fixed is.

Reply:
Ahhhhh. I'm starting to remember now. If memory serves me correctly, there are files named `version` with no extension in the Windows code base that sometimes got picked up instead either because they took priority in the include path or because the project doesn't use STL/a newer STL version. These files - from what I remember - aren't even code files and cause such projects to fail compilation. Doing a cpp version check is _probably_ okay here, even if not ideal

Reply:
Hopefully nobody has the idea of making `bit` files in the same fashion 😅